contracts/FlowIDTableStaking.cdc

```diff
@@ -1324,7 +1337,12 @@ access(all) contract FlowIDTableStaking {
 
                 // Iterate through all the withheld node's delegators
                 // and calculate their decreased rewards as well
-                for delegator in nodeRecord.delegators.keys {
+                let keys = nodeRecord.delegators.keys
```

Comment:
Why this change? Did we get rid of for...in?

Reply:
This is due to references are being not allowed in the for-loop. With the new [member-access semantics change](https://github.com/onflow/flips/blob/120b8aa473e5040db2b99034d2118fb78d053cdf/cadence/20230517-member-access-semnatics.md), the `nodeRecord.delegators.keys` now returns `&[T]`, and is unable to directly use in the loop.

This can be improved: https://github.com/onflow/cadence/issues/2784, but for now, we'll have to use the `while` loop in the meantime.

Reply:
Maybe open a tech debt issue that reverts these changes/this code regression once the Cadence improvements for working with references has landed (dereferencing a value reference, looping over references to arrays, e.g. onflow/cadence#2784)

Reply:
Done: https://github.com/onflow/flow-core-contracts/issues/385